Modelica_DeviceDrivers/Blocks/Communication.mo

```diff
@@ -110,26 +107,24 @@ provided by the parameter <b>memoryID</b>. If the shared memory partition does n
     parameter Integer port_recv=10001
       "Listening port number of the server. Must be unique on the system"
       annotation (Dialog(group="Incoming data"));
-    Modelica_DeviceDrivers.Blocks.Interfaces.PackageOut pkgOut(pkg = SerialPackager(bufferSize))
-                                       annotation (Placement(transformation(
+    Modelica_DeviceDrivers.Blocks.Interfaces.PackageOut pkgOut(pkg = SerialPackager(if autoBufferSize then bufferSize else userBufferSize))
+      annotation (Placement(transformation(
```

Comment:
Why replacing `bufferSize` by the conditional expression `if autoBufferSize then bufferSize else userBufferSize` (here and similarly at other places)?

`bufferSize` is already defined by the conditional expression
```modelica
 when initial() then
       bufferSize = if autoBufferSize then alignAtByteBoundary(pkgOut.autoPkgBitSize) else userBufferSize;
end when;
```
hence, that seems to introduce redundant code.